myserver/src/authStuff/signin.js

```diff
@@ -0,0 +1,71 @@
+import React, {Component} from 'react';
+import axios from 'axios';
+
+class Signin extends Component {
+    constructor(props){
+        super(props)
+        this.state = {
+            username: '',
+            password: '',
+            department: ''
+        }
+    }
+
+    fooFunction = (e) => {
+        console.log(`hi`)
+     }
+
+    render() {
+
+        return (
+            <div className="signin">
+            <p>hello world</p>
+            <form onSubmit={this.handleSubmit}>
+                <div>
+                    <label htmlFor="username">Username: </label>
+                    <input name="username" type="text" placeholder="Username" onChange={this.inputHandler}/>
+                </div>
+     
+                <div>
+                    <label htmlFor="password">Password: </label>
+                    <input name="password" type="text" placeholder="Password" onChange={this.inputHandler}/>
+                </div>
+
+                <div>
+                    <label htmlFor="department">Department: </label>
+                    <input name="department" type="text" placeholder="Department" onChange={this.inputHandler}/>
+                </div>
```

Comment:
All of your input/label divs look just about the same. Why not try to make them into a component? Here's how you could do it by just having an extra method on this component (since you aren't using this pattern elsewhere):
```javascript
makeInputDiv = (name, type='text') => {
  const capializedName = name[0].toUpperCase() + name.slice(1);
  return (
    <div>
      <label htmlFor={name}>
        {capializedName}: 
      </label>
      <input
        name={name} 
        type={type} 
        placeholder={capializedName} 
        onChange={this.inputHandler}
      />
    </div>
  );
};
```

Reply:
and the usage in your render method's return:
```javascript
// ...
<form onSubmit={this.handleSubmit}>
  {this.makeInputDiv('username')}
  {this.makeInputDiv('password')}
  {this.makeInputDiv('department')}
// ...
```
or, more programmatic:
```javascript
// ...
<form onSubmit={this.handleSubmit}>
  {[ 'username', 'password', 'department' ]
    .map(name => this.makeInputDiv(name))
  }
// ...
```